### PR TITLE
Make locale errors from translatable type work with locale

### PR DESCRIPTION
**Pull request by JevgenijVisockij**
…e FormattableTextArea compatible with translatableType

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | This PR is meant to make locale errors from translatable type work with simplified forms as well as make FormattedTextareaType work with TranslatableType
| Type?         | refacto
| Category?     | BO
| BC breaks?    | yes
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | This PR best to be tested with pr to example-modules(https://github.com/PrestaShop/example-modules/pull/28/).That PR contains simplified form with translatable fields and errors. Please test the form behaves in a relevant way.<br/><br/>Also some of the already existing translatable fields need to be tested to make sure they were not affected. Please test 1 or 2 BO page with fields that can be translated to check the form errors are working as expected.

BC break because TranslatableType now extends TranslatorAwareType so any module that extended TranslatableType before might run into problems. 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21062)

### CODE REVIEW

**Review by matks — DISMISSED** [review]
Code is OK but we need to look at one or two migrated pages using TranslatableType to be sure that these changes do not affect badly how errors are displayed today.

We might find that we need to update a little every page already using TranslatableType

File: src/PrestaShopBundle/Form/Admin/Type/TranslatableType.php [hunk 1]
```diff
@@ -116,6 +122,31 @@ public function buildForm(FormBuilderInterface $builder, array $options)
      */
     public function buildView(FormView $view, FormInterface $form, array $options)
     {
+        $errors = [];
+
+        foreach ($view->vars['errors'] as $existingError) {
+            $errors[] = $existingError;
```
Comment (sowbiba) [inline src/PrestaShopBundle/Form/Admin/Type/TranslatableType.php]: based on https://github.com/symfony/symfony/blob/v3.4.43/src/Symfony/Component/Form/FormView.php
index `errors` can be undefined in `$view->vars` array

If it's defined and is an array, why not `$errors = $view->vars['errors']`

So can you try `$errors = $view->vars['errors'] ?? [];`

Comment (JevgenijVisockij) [reply src/PrestaShopBundle/Form/Admin/Type/TranslatableType.php]: This makes sense, thanks!

Comment (JevgenijVisockij) [reply src/PrestaShopBundle/Form/Admin/Type/TranslatableType.php]: Ahh nevermind. $view->vars['errors'] is an iterator not an array. And it won't allow me to edit it, thats why in line after it there is foreach trough that itterator to get errors one by one.

File: src/PrestaShopBundle/Resources/views/Admin/macros.html.twig [hunk 2]
```diff
@@ -142,11 +142,6 @@
     {% endfor %}
 
     {#iterating over child errors because errors can be nested#}
```
Comment (JevgenijVisockij) [inline src/PrestaShopBundle/Resources/views/Admin/macros.html.twig:144]: Reasoning for that is that now I am collecting all the errors from children in TranslatableType so this is no longer needed. As far as I tested translatable types still work and according due commits and comment this was added specifically for translatable type, but I am not 100% sure this won't affect something else.

Comment (matks) [reply src/PrestaShopBundle/Resources/views/Admin/macros.html.twig:144]: Well, QA will be able to help us asserting the 100% 😉

**Review by matks — DISMISSED** [review]

File: src/PrestaShopBundle/Form/Admin/Type/TranslatableType.php (hunk 1 above)
Comment (PierreRambaud) [reply src/PrestaShopBundle/Form/Admin/Type/TranslatableType.php]: you can use ̀iterator_to_array`or try to cast it to array ?

File: src/PrestaShopBundle/Form/Admin/Type/TranslatableType.php [hunk 3]
```diff
@@ -116,6 +122,31 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
+        }
+
+        $errorsByLocale = $this->getErrorsByLocale($view, $form, $options['locales']);
+
+        if (!empty($errorsByLocale)) {
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Form/Admin/Type/TranslatableType.php]:
```suggestion
        if ($errorsByLocale !== null) {
```

File: src/PrestaShopBundle/Form/Admin/Type/TranslatableType.php (hunk 1 above)
Comment (JevgenijVisockij) [reply src/PrestaShopBundle/Form/Admin/Type/TranslatableType.php]: iterator_to_array to array worked perfectly, thanks!

**Review by matks — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by Robin-Fischer-PS** [comment]
Hi @JevgenijVisockij !

I have an error on the "demosymfonyform" module : When I fill the fields with valid values, if I save, there is an exception : 

pr21062 exception module

However, I think this is only related to the module. Otherwise the PR seems OK for me, so if you can confirm that the error comes from the module and not the PR, I will validate your PR :)

**Comment by JevgenijVisockij** [comment]
> Hi @JevgenijVisockij !
> 
> I have an error on the "demosymfonyform" module : When I fill the fields with valid values, if I save, there is an exception :
> 
> pr21062 exception module
> 
> However, I think this is only related to the module. Otherwise the PR seems OK for me, so if you can confirm that the error comes from the module and not the PR, I will validate your PR :)

Hey thanks for testing!

Yes can confirm that error is strictly module related.

**Comment by Robin-Fischer-PS** [comment]
Thanks @JevgenijVisockij , then it's QA :heavy_check_mark: :)

**Comment by Progi1984** [comment]
Thanks @JevgenijVisockij & @Robin-Fischer-PS

**Pull request merged**